addons/stock_account/models/product.py

```diff
@@ -81,7 +81,7 @@ def _get_product_accounts(self):
         accounts.update({
             'stock_input': res['stock_input'] or self.categ_id.property_stock_account_input_categ_id,
             'stock_output': res['stock_output'] or self.categ_id.property_stock_account_output_categ_id,
-            'stock_valuation': self.categ_id.property_stock_valuation_account_id or False,
+            'stock_valuation': self.categ_id.property_stock_valuation_account_id,
```

Comment:
Yeah, just keeping the type the same.
These `emptyrecord or None` and `emptyrecord or False` are annoying because they force us to manage multiple types and use defensive clauses everywhere when it could be as simple as doing nothing...

Reply:
If invoice is a new record, invoice.fiscal_position_id will be a new record too

Reply:
But that's fine?!

Reply:
@william-andre It works with new records indeed. Forget my comment.